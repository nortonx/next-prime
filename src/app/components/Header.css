header nav ul {
  display: flex;
  list-style: none;
}

header nav ul li a {
  text-decoration: none;
}
