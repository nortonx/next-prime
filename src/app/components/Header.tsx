"use client";

import Link from "next/link";
import "./Header.css";
import { Button } from "primereact/button";
const Header = () => {
  return(
    <header data-testid="header" className="flex w-full">
      <nav className="flex w-full align-items-center justify-content-between">
        <div className="brand ml-3">Brand</div>
        <ul className="flex mr-3" >
          <li className="mr-2">
            <Link href="/password-generator">
              <Button
                label="Password Generator"
                icon="pi pi-lock"
              />
            </Link>
          </li>
          <li className="mr-2">
            <Link href="/repositories">
              <Button
                label="Repositories"
                icon="pi pi-github"
              />
            </Link>
          </li>
          <li className="mr-2">
            <Link href="/word-count">
              <Button
                label="Word Count"
                icon="pi pi-check-circle"
              />
            </Link>
          </li>
          <li className="mr-2">
            <Link href="/about">
              <Button
                label="About"
                icon="pi pi-info-circle"
              />
            </Link>
          </li>
        </ul>
      </nav>
    </header>
  )
}

export default Header;