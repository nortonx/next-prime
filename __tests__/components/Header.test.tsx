import { render, screen } from "@testing-library/react"
import Header from "@/app/components/Header"

describe("Header", () => {
  it("should render and match snapshot", async () => {
    const { container } = render(<Header/>)
    const page = await screen.findByTestId("header");
    expect(page).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})