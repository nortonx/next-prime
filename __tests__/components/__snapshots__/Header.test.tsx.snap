// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render and match snapshot 1`] = `
<div>
  <header
    class="flex w-full"
    data-testid="header"
  >
    <nav
      class="flex w-full align-items-center justify-content-between"
    >
      <div
        class="brand ml-3"
      >
        Brand
      </div>
      <ul
        class="flex mr-3"
      >
        <li
          class="mr-2"
        >
          <a
            href="/password-generator"
          >
            <button
              aria-label="Password Generator"
              class="p-button p-component"
              data-pc-name="button"
              data-pc-section="root"
            >
              <span
                class="p-button-icon p-c p-button-icon-left pi pi-lock"
                data-pc-section="icon"
              />
              <span
                class="p-button-label p-c"
                data-pc-section="label"
              >
                Password Generator
              </span>
            </button>
          </a>
        </li>
        <li
          class="mr-2"
        >
          <a
            href="/repositories"
          >
            <button
              aria-label="Repositories"
              class="p-button p-component"
              data-pc-name="button"
              data-pc-section="root"
            >
              <span
                class="p-button-icon p-c p-button-icon-left pi pi-github"
                data-pc-section="icon"
              />
              <span
                class="p-button-label p-c"
                data-pc-section="label"
              >
                Repositories
              </span>
            </button>
          </a>
        </li>
        <li
          class="mr-2"
        >
          <a
            href="/word-count"
          >
            <button
              aria-label="Word Count"
              class="p-button p-component"
              data-pc-name="button"
              data-pc-section="root"
            >
              <span
                class="p-button-icon p-c p-button-icon-left pi pi-check-circle"
                data-pc-section="icon"
              />
              <span
                class="p-button-label p-c"
                data-pc-section="label"
              >
                Word Count
              </span>
            </button>
          </a>
        </li>
        <li
          class="mr-2"
        >
          <a
            href="/about"
          >
            <button
              aria-label="About"
              class="p-button p-component"
              data-pc-name="button"
              data-pc-section="root"
            >
              <span
                class="p-button-icon p-c p-button-icon-left pi pi-info-circle"
                data-pc-section="icon"
              />
              <span
                class="p-button-label p-c"
                data-pc-section="label"
              >
                About
              </span>
            </button>
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
